### ISSUE TITLE
Testing broken on OSX

### ISSUE THREAD

**Issue by flyingfrog81** [discos/discos-backend#1]
Testing is not working in OSX. Tried with OSX version 10.6.8 and Python 2.7.2

```bash
INFO:discosbackend.server:Protocol handler: <class 'discosbackend.handlers.DBProtocolHandler'>
Traceback (most recent call last):
  File "test/run_simulator_server.py", line 11, in <module>
    DBProtocolHandler(Backend()))
  File "/Users/flyingfrog/.virtualenvs/discosbackend/lib/python2.7/site-packages/discosbackend/server.py", line 11, in run_server
    reactor.listenTCP(port, DBFactory(handler = handler))
  File "/Library/Frameworks/Python.framework/Versions/7.1/lib/python2.7/site-packages/twisted/internet/posixbase.py", line 419, in listenTCP
    p.startListening()
  File "/Library/Frameworks/Python.framework/Versions/7.1/lib/python2.7/site-packages/twisted/internet/tcp.py", line 867, in startListening
    raise CannotListenError, (self.interface, self.port, le)
twisted.internet.error.CannotListenError: Couldn't listen on any:8988: [Errno 48] Address already in use.
^C
======================================================================
```

**Comment by flyingfrog81** [comment]
It seems like the problem is related to the killing of the server process which is not executed. The **subprocess** python module should spawn a *kill* instruction to the proper *PID* . Maybe the *PID* is wrong or maybe it's not correctly executed, but it is working on every linux version we tried. The instruction is coded like this:

```python
subprocess.call(["kill", "-9", str(pid)])
```

(see https://github.com/discos/discos-backend/blob/59bd3a7/test/test_always_ok_server.py#26 )

**Comment by flyingfrog81** [comment]
As in commit 61363200e the issue was solved in three steps:

1. **subprocess32** was imported replacing the old *subprocess* module which seems to be deprecated on python2 . 
2. instead of closing a process by its PID we used the process object with its **terminate** method
3. We augmented the sleep time between server time and test execution as on slower PCs the server was not running when the client tried to access it result in no connection.
